src/reducers/transactionReducer.ts

```diff
@@ -165,6 +167,23 @@ export default function transactionReducer(state: TransactionState, action: Tran
         estimatedFeeError: action.payload.estimatedFeeError,
         estimatedFeeLoading: action.payload.estimatedFeeLoading
       };
+    case TransactionActionTypes.SET_TRANSFER_AMOUNT: {
+      const { transferAmount, balance, chainDecimals } = action.payload;
+      let [actualValue, message] = evalUnits(transferAmount, chainDecimals);
+      if (
+        state.estimatedFee &&
+        actualValue &&
+        new BN(balance).sub(actualValue).add(new BN(state.estimatedFee)).isNeg()
+      ) {
+        actualValue = null;
+        message = 'Not enough funds for this transaction.';
```

Comment:
I recommended to move this to the reducer in order to try to avoid logic on the component level.

Reply:
Well, then it's broken because balance is not being updated (not part of the reducer) and if we want to do it properly we will again run into a situation where displaying a particular component is necessary for the reducer logic to work correctly (cause that component will have `useEffect` that will keep updating balance in the *global* reducer).